aten/src/ATen/native/cuda/Normalization.cu

```diff
@@ -174,12 +190,12 @@ __device__ scalar_t reduce(Op op, PTA tensor, int plane) {
 
 template <typename scalar_t, typename accscalar_t, bool train>
 __global__ void batch_norm_transform_input_kernel(
-    const PackedTensorAccessor<scalar_t, 3, at::RestrictPtrTraits> input,
-    PackedTensorAccessor<scalar_t, 3, at::RestrictPtrTraits> output,
-    const PackedTensorAccessor<typename std::conditional<train, accscalar_t, scalar_t>::type, 1, at::RestrictPtrTraits> mean_,
-    const PackedTensorAccessor<typename std::conditional<train, accscalar_t, scalar_t>::type, 1, at::RestrictPtrTraits> var_or_invstd,
-    const PackedTensorAccessor<scalar_t, 1, at::RestrictPtrTraits> weight,
-    const PackedTensorAccessor<scalar_t, 1, at::RestrictPtrTraits> bias,
+    const PackedTensorAccessor<scalar_t, 3> input,
```

Comment:
I'm kinda curious how much does `__restrict__` gain us here.

Reply:
If it's non-trivial, we might need to consider make it work for saved_mean and saved_invstd.